test/simplify_qdq_test.cpp

```diff
@@ -0,0 +1,537 @@
+#include <migraphx/simplify_qdq.hpp>
+#include <migraphx/program.hpp>
+#include <migraphx/ref/target.hpp>
+#include <migraphx/instruction.hpp>
+#include <test.hpp>
+#include <migraphx/make_op.hpp>
+#include <migraphx/dead_code_elimination.hpp>
+#include <migraphx/pass_manager.hpp>
+#include <migraphx/matcher.hpp>
+
+bool is_convolution(const migraphx::instruction& ins) { return ins.name() == "convolution"; }
+bool is_dot(const migraphx::instruction& ins) { return ins.name() == "dot"; }
+
+void run_pass(migraphx::module& m)
+{
+    migraphx::simplify_qdq sqdq;
+    sqdq.apply(m);
+}
+
+migraphx::instruction_ref add_quantize_op(migraphx::module& m,
+                                          const std::string& name,
+                                          migraphx::instruction_ref x,
+                                          migraphx::instruction_ref scale,
+                                          migraphx::instruction_ref shift)
+{
+    auto lens = x->get_shape().lens();
+    auto scale_mb =
+        m.add_instruction(migraphx::make_op("multibroadcast", {{"output_lens", lens}}), scale);
+    auto shift_mb =
+        m.add_instruction(migraphx::make_op("multibroadcast", {{"output_lens", lens}}), shift);
+    return m.add_instruction(migraphx::make_op(name), x, scale_mb, shift_mb);
+}
+
+TEST_CASE(dot)
+{
+    migraphx::shape sh1{migraphx::shape::float_type, {1280, 1000}};
+    migraphx::shape sh2{migraphx::shape::float_type, {1000, 1024}};
+    migraphx::shape ss{migraphx::shape::float_type, {1}};
+    migraphx::shape zs{migraphx::shape::int8_type, {1}};
+
+    std::vector<float> scale{0.5};
+    std::vector<int> zero{0};
+
+    migraphx::module m1;
+    {
+        auto t1 = m1.add_parameter("t1", sh1);
+        auto t2 = m1.add_parameter("t2", sh2);
+        auto sc = m1.add_literal(ss, scale);
+        auto z  = m1.add_literal(zs, zero);
+
+        auto q1 = add_quantize_op(m1, "quantizelinear", t1, sc, z);
+        auto d1 = add_quantize_op(m1, "dequantizelinear", q1, sc, z);
+        auto q2 = add_quantize_op(m1, "quantizelinear", t2, sc, z);
+        auto d2 = add_quantize_op(m1, "dequantizelinear", q2, sc, z);
+        auto dot =
+            m1.add_instruction(migraphx::make_op("dot", {{"alpha", 1}, {"beta", 0}}), d1, d2);
```

Comment:
Shouldn't there be a `quantizelinear` after the dot operator?

Reply:
No, there is no quantizelinear after the dot operator